app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackPresenter.kt

```diff
@@ -37,40 +40,64 @@ class FeedbackPresenter(
         }
     }
 
-    override fun setClassifications(classifications: List<Classifier.Classification>) {
 
-        results.clearAndAddAll(classifications)
+    override fun setClassifications(classifications: LinkedHashMap<String, ArrayList<Classifier.Classification>>) {
+        results.clear()
+        results.putAll(classifications)
 
-        if (results.isEmpty()) {
+        val labels: LinkedList<Pair<String, Int>> = LinkedList()
+
+        classifications.entries.forEach { entry ->
+
+            val list = entry.value
+            if (list.isEmpty()) {
+                labels.add(Pair.create("", 0))
+                return
+            }
+
+            if (!list[0].hasTitle()) {
+                labels.add(Pair.create("", 0))
+                return
+            }
+
+            labels.add(Pair.create(list[0].title, formatConfidence(list[0].confidence)))
+        }
+
+        view?.showMainClassifications(labels)
+        
+
+        oldResults.clearAndAddAll(classifications.entries.first().value)
+
+        if (oldResults.isEmpty()) {
             view?.showNoClassifications()
             return
         }
 
         if (userFeedback == null) {
-            view?.showClassifications(results.head().title, results.tail())
+            view?.showClassifications(oldResults.head().title, oldResults.tail())
```

Comment:
Why oldResults instead of results (here and in the lines below)?

Reply:
To be able to run the app without full refactory. This should be fixed on Feedback task

Reply:
Ok